client/state/reader/streams/reducer.js

```diff
@@ -57,6 +61,18 @@ export const items = ( state = [], action ) => {
 
 		case READER_STREAMS_SHOW_UPDATES:
 			return [ ...action.payload.items, ...state ];
+		case READER_STREAMS_DISMISS_POST: {
+			const postKey = action.payload.postKey;
+			const indexToRemove = findIndex( state, item => keysAreEqual( item, postKey ) );
+
+			if ( indexToRemove === -1 ) {
+				return state;
+			}
+
+			const newState = [ ...state ];
+			newState[ indexToRemove ] = newState.pop(); // set the dismissed post location to the last item from the recs stream
```

Comment:
the comment doesn't seem to jive with what this does. it replaces the dismissed item with the last item in the stream, ya?

Reply:
is that what the old code did?

Reply:
yea, that comment is unclear. better comment would be: "replace the dismissed item with the last item"